src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2330,6 +2451,7 @@ ExtractInsertValuesList(Query *query, Var *partitionColumn)
  * by setting citus.enable_router_execution flag to false.
  */
 static bool
+/* TODO Does the command and implementation agree with each other ? */
```

Comment:
Did we address this comment?

Reply:
We may completely remove this function. Modify commands were already handled by CreateInsertSelectPlan and CreateModifyPlan. Also, we check whether a SELECT query is router plannable with CreateSingleTaskRouterPlan. We may also remove TODO declared in the CreateRouterPlan.

Reply:
Well, I'm not sure a future reader will be able to parse this TODO, so maybe remove the "and it has a filter to reduce number of shard pairs to one, all shard pairs are located on the same node" bit from the function comment and remove this TODO.